lib/engine-addon.js

```diff
@@ -387,10 +387,76 @@ module.exports = {
         return mergeTrees([
           externalTree,
           engineJSTree,
-          engineCSSTree
+          compiledEngineCSSTree
         ].filter(Boolean), { overwrite: true });
       };
 
+      this.compileLazyEngineStyles = function compileLazyEngineStyles(vendorTree, externalTree) {
+        var vendorCSSTree = buildVendorCSSTree.call(this, vendorTree);
+        var engineStylesOutputDir = 'engines-dist/' + this.name + '/assets/';
+
+        // Get base styles tree.
+        var engineStylesTree = this._treeFor('addon-styles');
+
+        var primaryStyleTree = null;
+        if (engineStylesTree) {
```

Comment:
> I suspect it may have to do with actually needing to keep vendor + add-on style separate?

Confirm, that is exactly right. If we merge them, then do the `concat` for `**/*.css` we loose the ability to actually have separate `engine.css` and `engine-vendor.css`.